istioctl/cmd/waypoint.go

```diff
@@ -191,15 +195,15 @@ func waypointCmd() *cobra.Command {
   istioctl x waypoint list -A`,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			writer := cmd.OutOrStdout()
-			err := getKubeClient()
+			kubeClient, err := kubeClientWithRevision(ctx, "")
 			if err != nil {
 				return fmt.Errorf("failed to create Kubernetes client: %v", err)
 			}
-			ns := defaultNamespace
+			ns := ctx.DefaultNamespace()
 			if allNamespaces {
 				ns = ""
-			} else if namespace != "" {
-				ns = namespace
+			} else if ctx.Namespace() != "" {
+				ns = ctx.Namespace()
 			}
```

Comment:
nit: add NamespaceOrDefault into `ctx` or similar.

Can then replace all of `handlers.HandleNamespace` with it as well

Reply:
Done.